extensions/amp-story/0.1/amp-story-auto-ads.js

```diff
@@ -78,43 +90,84 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
     return Promise.resolve();
   }
 
+
   /**
    * build page and start preloading
    * @private
    */
   schedulePage_() {
-    const page = this.makeMockPage();
-    this.adElements_.push(page);
+    const page = this.createAdPage_();
+    this.adPageEls_.push(page);
 
     this.ampStory_.element.appendChild(page);
 
-    // TODO(ccordry): need to fake distance from page to force load
-
     page.getImpl().then(impl => {
       this.ampStory_.addPage(impl);
     });
   }
 
+
   /**
-   * temporary to be replaced with real fetching
+   * create an `amp-story-page` containing an `amp-ad`
+   * @private
    */
-  makeMockPage() {
-    const ampStoryAdPage = document.createElement('amp-story-page');
-    const id = this.adsPlaced_ + 1;
-    ampStoryAdPage.id = `i-amphtml-ad-page-${id}`;
-    ampStoryAdPage.setAttribute('ad', '');
-    ampStoryAdPage./*OK*/innerHTML = `
-      <amp-story-grid-layer template="vertical">
-        <h1>Ad Page #${id}</h1>
-        <p>This is ad #${id} shown in this story.</p>
-      </amp-story-grid-layer>
-      `;
+  createAdPage_() {
+    // TODO(ccordry) add new <amp-story-cta-layer>
+    const ampStoryAdPage = this.createPageElement_();
+    const ampAd = this.createAdElement_();
+
+    ampStoryAdPage.appendChild(ampAd);
+
+    this.iscurrentAdLoaded_ = false;
+
+    // set up listener for ad-loaded event
+    ampAd.getImpl().then(impl => {
+      const signals = impl.signals();
+      return signals.whenSignal(CommonSignals.INI_LOAD);
+    }).then(() => {
+      this.iscurrentAdLoaded_ = true;
+    });
+
     return ampStoryAdPage;
   }
 
 
   /**
-   * Get array containing all pages of associated story
+   * @return {!Element}
+   * @private
+   */
+  createPageElement_() {
+    const id = this.adsPlaced_ + 1;
```

Comment:
adsPlaced_ => adPagesCnt_

Reply:
Is it problematic that this function can return two different elements with the same ID? (since adsPlaced_ is not incremented here)

Reply:
I addressed both of these concerns by adding a new `adPagesCreated` which is incremented inside this method. I think this is the best solution because ads can be created but never placed due to bad fetch, not loading in time etc. so they should be tallied separately. Also this helps mapping the config ad number to the id (e.g.. template ad #2 is never placed, so template ad #3 gets and id of 2)